plugin/trino-hive/pom.xml

```diff
@@ -33,6 +33,11 @@
             <artifactId>trino-parquet</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>io.trino</groupId>
+            <artifactId>trino-parser</artifactId>
```

Comment:
Why are we including this?

Reply:
The `dependency-scope-maven-plugin` threw an error which I believe means that coral-trino now has a compile time dependency on the trino-parser.
```
[ERROR] Found a problem with test-scoped dependency io.trino:trino-parser
  Scope compile was expected by artifact: com.linkedin.coral:coral-trino:2.0.24
```

Reply:
> This is not added new but moved from a testing dependency.

so, effectively, new dependency

>  I believe means that coral-trino now has a compile time dependency on the trino-parser. 

as provided?

Reply:
The parser is not designed to be depended on by other projects. It’s not guaranteed to be stable across versions, etc. 

If Coral made the choice to depend on it, it should be shaded or re-namespaced. Otherwise, a change to the parser may break Coral and lock us into no being able to change the parser without breaking this integration.

Reply:
> If Coral made the choice to depend on it, it should be shaded or re-namespaced. 

that's ideal.

in the worst case, we should pick the `trino-parser` version Coral was compiled against and tested with, but that may be harder to do when `trino-parser` is present in the same maven reactor.

Reply:
I just raised an issue in the Coral project. This is not particularly urgent. Let's wait to see what they say.

Reply:
Thanks for bringing this up. We will look into it. Another option is to separate this dependency into a new module, but it will complicate the naming of the existing modules (since there will be two modules about Trino conversions). Is excluding that dependency when depending on Coral feasible?
For context, this dependency was added as part of an effort to convert Trino SQL to Coral IR.

Reply:
It is possible to add an exclusion for it, but it's not clear when it is safe to do so. How can we be certain that none of the code paths exercised by Trino depend on that library? Adding exclusions is generally an anti-pattern, only to be used as an escape hatch when you know exactly whether it's safe to do so.

Reply:
Actually, I do not understand the error `Found a problem with test-scoped dependency io.trino:trino-parser`. So in the [new version](https://github.com/linkedin/coral/blob/master/coral-trino/build.gradle) `trino-parser` is depended upon with `compile` scope. Where is `test` scope coming from?

Reply:
Ok. I think it is referring to this in `tino-hive/pom.xml`
``` 
       <dependency>
            <groupId>io.trino</groupId>
            <artifactId>trino-parser</artifactId>
            <scope>test</scope>
        </dependency>
```

Reply:
We make use of `trino-parser` as a test dependency for the Hive and Iceberg connectors. The issue is that a local dependency with a test scope will override a transitive dependency with compile scope (see [this](https://issues.apache.org/jira/browse/MNG-4156) for reference).

Reply:
I am taking a look at shading trino-parser in coral [Coral Issue#205: Shade trino-parser dependency of coral-trino](https://github.com/linkedin/coral/issues/205)

Reply:
after shading trino-parser we found IDE issues with shaded libraries, so for now we simply changed the trino-parser dependency into compileOnly/Implementation in https://github.com/linkedin/coral/pull/222 and this is already merged

Reply:
Coral 2.0.44 is released and can be adopted by Trino
CC @wmoustafa  @joshthoward